nexus/src/nexus.rs

```diff
@@ -1292,20 +1298,19 @@ impl Nexus {
         instance_name: &Name,
         disk_name: &Name,
     ) -> UpdateResult<db::model::Disk> {
-        // TODO: This shouldn't be looking up multiple database entries by name,
-        // it should resolve names to IDs first.
-        let authz_project = self
-            .db_datastore
-            .project_lookup_by_path(organization_name, project_name)
-            .await?;
-        let (authz_disk, db_disk) = self
-            .db_datastore
-            .disk_fetch(opctx, &authz_project, disk_name)
-            .await?;
-        let (authz_instance, db_instance) = self
-            .db_datastore
-            .instance_fetch(opctx, &authz_project, instance_name)
-            .await?;
+        let (_, authz_project, authz_disk, db_disk) =
+            LookupPath::new(opctx, &self.db_datastore)
```

Comment:
The convention that I'm seeing here is:

- We can *start* a lookup either by a "root resource name" (e.g., Organization) or by any UUID
- Once we have a resource, we can look up contained resources by name
- Accessing the resources is combined with the authz check

From an API point-of-view, I like this pattern!